node/node_test.go

```diff
@@ -342,7 +358,64 @@ func TestNode_GetKeyValuePairs(t *testing.T) {
 	assert.Equal(t, hex.EncodeToString(v2), resV2)
 }
 
+func TestNode_GetKeyValuePairs_GetAllLeavesShouldFail(t *testing.T) {
+	t.Parallel()
+
+	acc, _ := state.NewUserAccount([]byte("newaddress"))
+
+	accDB := &stateMock.AccountsStub{}
+
+	expectedErr := errors.New("expected err")
+	acc.SetDataTrie(
+		&trieMock.TrieStub{
+			GetAllLeavesOnChannelCalled: func(leavesChannels *common.TrieIteratorChannels, ctx context.Context, rootHash []byte, _ common.KeyBuilder) error {
+				go func() {
+					close(leavesChannels.LeavesChan)
```

Comment:
this is a flaky test (might fail from time to time)
Reason? You close the leaves chan before writing the error in the error chan. Since this is done on a go routine, we might hit the case when the LeavesChan is closed but no error written yet in ErrChan. 
```GO
assert.Equal(t, expectedErr, err)
```
might fail

Suggestion here and perhaps in other tests (if applicable) is to revert the instructions order from L373-L343
The production implementation is done correctly :+1:

Reply:
thanks :+1: changed in all places